### PR TITLE
libdrm: build with meson

### PR DESCRIPTION
**Pull request by colemickens**
###### Motivation for this change

I'm testing this jointly with #56199 and assume it might be staged with it?

I'm doing this as the next libdrm release seemed to require it, similar to the next release of mesa.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/libdrm/default.nix [hunk 1]
```diff
@@ -25,13 +28,15 @@ stdenv.mkDerivation rec {
   preConfigure = stdenv.lib.optionalString stdenv.isDarwin
     "echo : \\\${ac_cv_func_clock_gettime=\'yes\'} > config.cache";
 
-  configureFlags = [ "--enable-install-test-programs" ]
+  mesonFlags =
+    [ "-Dinstall-test-programs=true" ]
     ++ stdenv.lib.optionals (stdenv.isAarch32 || stdenv.isAarch64)
       [ "--enable-tegra-experimental-api" "--enable-etnaviv-experimental-api" ]
-    ++ stdenv.lib.optional stdenv.isDarwin "-C"
```
Comment (colemickens) [inline pkgs/development/libraries/libdrm/default.nix]: please review this deletion. I'm not sure what it was for, or what it's meson equivalent is.

**Comment by colemickens** [comment]
cc: @vcunat @dtzWill @eadwu

File: pkgs/development/libraries/libdrm/default.nix [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
-    url = "https://dri.freedesktop.org/libdrm/${name}.tar.bz2";
-    sha256 = "08yimlp6jir1rs5ajgdx74xa5qdzcqahpdzdk0rmkmhh7vdcrl3p";
+  name = "libdrm-${version}";
+  version = "2.4.97";
+  rev = "${name}";
+  sha256 = "065w0zyyfv3yzlw76pvsdjrkxd46wp5cwv472766is0mbi212rdp";
+
+  src = fetchgit {
+    url = "https://gitlab.freedesktop.org/mesa/drm";
+    inherit rev sha256;
```
Comment (dtzWill) [inline pkgs/development/libraries/libdrm/default.nix]: Why the change of source?

File: pkgs/development/libraries/libdrm/default.nix [hunk 3]
```diff
@@ -25,13 +28,15 @@ stdenv.mkDerivation rec {
   preConfigure = stdenv.lib.optionalString stdenv.isDarwin
     "echo : \\\${ac_cv_func_clock_gettime=\'yes\'} > config.cache";
```
Comment (dtzWill) [inline pkgs/development/libraries/libdrm/default.nix]: This should be removed-- if we're using meson it does nothing :).

File: pkgs/development/libraries/libdrm/default.nix [hunk 4]
```diff
@@ -25,13 +28,15 @@ stdenv.mkDerivation rec {
...
 
-  configureFlags = [ "--enable-install-test-programs" ]
+  mesonFlags =
+    [ "-Dinstall-test-programs=true" ]
     ++ stdenv.lib.optionals (stdenv.isAarch32 || stdenv.isAarch64)
       [ "--enable-tegra-experimental-api" "--enable-etnaviv-experimental-api" ]
```
Comment (dtzWill) [inline pkgs/development/libraries/libdrm/default.nix]: `"-Dtegra=true" "-Detnaviv=true"` , at least that's what I came up with :).

File: pkgs/development/libraries/libdrm/default.nix (hunk 1 above)
Comment (dtzWill) [reply pkgs/development/libraries/libdrm/default.nix]: It can be dropped as well, it's asking configure to use the cache ( https://www.gnu.org/software/autoconf/manual/autoconf-2.65/html_node/Cache-Files.html ), presumably to pick up the value put there in `preConfigure` previously.

Comment (dtzWill) [reply pkgs/development/libraries/libdrm/default.nix]: Not sure that the `clock_gettime` workaround is still useful (meson or no), may need some testing or investigation on that.

**Review by dtzWill — CHANGES_REQUESTED** [review]
Thanks! Agreed we'll need to move this to meson "soon" anyway.

**Comment by dtzWill** [comment]
Also, and maybe this is why I was asked to review ;), previously: https://github.com/NixOS/nixpkgs/pull/48218 .  Been using it ever since, so far so good ^_^.

File: pkgs/development/libraries/libdrm/default.nix (hunk 2 above)
Comment (colemickens) [reply pkgs/development/libraries/libdrm/default.nix]: It makes it easier to point at git revisions, but maybe not appropriate for this PR. I'll revert.

**Comment by colemickens** [comment]
@dtzWill Thank you for the review. I actually hadn't seen that PR -- please let me know if you'd prefer to reopen your PR and drive it yourself. Otherwise, I'm happy to keep pushing this with your guidance.

**Comment by dtzWill** [comment]
On Sun, 17 Mar 2019 13:12:44 -0700, Cole Mickens <notifications@github.com> wrote:
> @dtzWill Thank you for the review. I actually hadn't seen that PR -- please let me know if you'd prefer to reopen your PR and drive it yourself. Otherwise, I'm happy to keep pushing this with your guidance.

I ran out of steam pushing that PR, so probably not wise to put our
hopes there.  If you don't mind I think we can wrap this up here!

**Comment by colemickens** [comment]
I've updated this per the requested changes. Note, I don't have Darwin devices, so I can't test anything this affects in Darwin.

**Comment by colemickens** [comment]
Rebased, pushed. 19.1's release is coming very soon. Shall we try to get this and [the mesa/meson change](https://github.com/NixOS/nixpkgs/pull/56199) pushed?

**Comment by colemickens** [comment]
Rebased on and re-pointed at staging. cc: @grahamc

**Comment by nh2** [comment]
This broke the build with musl:

https://github.com/NixOS/nixpkgs/issues/66441
